crates/syn-solidity/src/yul/ident/path.rs

```diff
@@ -0,0 +1,61 @@
+use crate::{Spanned, YulIdent};
+
+use proc_macro2::Span;
+use std::fmt;
+use syn::{
+    parse::{Parse, ParseStream},
+    punctuated::Punctuated,
+    Result, Token,
+};
+
+/// In inline assembly, only dot-less identifiers can be declared, but dotted
+/// paths can reference declarations made outside the assembly block.
+///
+/// Solidity Reference:
+/// <https://docs.soliditylang.org/en/latest/grammar.html#a4.SolidityParser.yulPath>
+#[derive(Clone)]
+pub enum YulPath {
+    /// Dotless path, references a declared varaible inside the assembly block.
+    SimplePath(YulIdent),
+
+    /// Dotted path, references a declared variable outside the assembly block.
+    DottedPath(Punctuated<YulIdent, Token![.]>),
+}
```

Comment:
This should just be a wrapper struct for `Punctuated<YulIdent, Token![.]>`, the enum seems unnecessary